msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
> We need to percolate colours up the tree, 

This should be done by the client. If we just give the ability to colour branches, the client code can decide what to do. What you're suggesting is too specific to the current use-case. People might want to use this for all sorts of things, and by making this over-complicated and over-specific you exclude this. Keep it simple: an optional colour, per branch. Nothing else.